### PR TITLE
TMem - no base class

### PR DESCRIPTION
**Pull request by bergloman**

### CODE REVIEW

File: src/glib/base/dt.cpp [hunk 1]
```diff
@@ -296,13 +296,12 @@ void TMem::Resize(const int& _MxBfL){
...
-  if (Bf!=NULL && Owner){delete[] Bf;}
-  Owner = true;
+  if (Bf!=NULL){delete[] Bf;}
   Bf=NewBf;
 }
 
-TMem::TMem(const TStr& Str) : TMemBase() {
-	MxBfL = Str.Len(); BfL = MxBfL; Bf = NULL; Owner = true;
+TMem::TMem(const TStr& Str) {
+	MxBfL = Str.Len(); BfL = MxBfL; Bf = NULL;
```
Comment (blazf) [inline src/glib/base/dt.cpp]: Please use spaces instead of tabs

File: src/glib/base/dt.h [hunk 2]
```diff
@@ -175,44 +176,63 @@ class TMem;
...
+  ~TMem() { if (Bf != NULL) { delete[] Bf; }; Bf = NULL; }
   explicit TMem(TSIn& SIn) {
     SIn.Load(MxBfL); SIn.Load(BfL);
-    Bf = new char[MxBfL = BfL]; SIn.LoadBf(Bf, BfL); Owner = true; }
+    Bf = new char[MxBfL = BfL]; SIn.LoadBf(Bf, BfL); }
+
+  int Len() const { return BfL; }
+  bool Empty() const { return BfL == 0; }
+  char* GetBf() const { return Bf; }
+  void Copy(const TMem& Mem) {
```
Comment (blazf) [inline src/glib/base/dt.h]: Can we move multiline functions to .cpp file?

File: src/qminer/qminer_storage.cpp [hunk 3]
```diff
@@ -956,6 +956,77 @@ void TRecSerializator::SetFixedJsonVal(TMemBase& RecMem,
     SetFixedJsonVal(RecMem.GetBf(), RecMem.Len(), FieldSerialDesc, FieldDesc, JsonVal);
 }
 
+
+void TRecSerializator::SetFieldByte(TMem& RecMem,
```
Comment (blazf) [inline src/qminer/qminer_storage.cpp]: Can we remove duplicates by using templates for `TMem` and `TMemBase`?

**Comment by blazf** [comment]
@rupnikj can you also give a look at the changes, since they are quite fundamental?

File: src/glib/base/dt.cpp (hunk 1 above)
Comment (bergloman) [reply src/glib/base/dt.cpp]: Fixed. The file is filled with mixed indenting anyway...

File: src/glib/base/dt.h (hunk 2 above)
Comment (bergloman) [reply src/glib/base/dt.h]: Done for the more complex ones. 2-liners are still in the .h file

File: src/qminer/qminer_storage.cpp (hunk 3 above)
Comment (bergloman) [reply src/qminer/qminer_storage.cpp]: Done. Take a look.

**Pull request merged**